src/Functions/h3PointDistRads.cpp

```diff
@@ -0,0 +1,129 @@
+        {
+            const auto * arg = arguments[i].get();
+            if (!WhichDataType(arg).isFloat64())
+                throw Exception(
+                        ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                        "Illegal type {} of argument {} of function {}. Must be Float64",
+                        arg->getName(), i, getName());
+        }
+        return std::make_shared<DataTypeFloat64>();
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        const auto * col_lat1 = checkAndGetColumn<ColumnFloat64>(arguments[0].column.get());
+        if (!col_lat1)
+            throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "Illegal type {} of argument {} of function {}. Must be Float64",
+                    arguments[0].type->getName(),
+                    1,
+                    getName());
+        const auto & data_lat1 = col_lat1->getData();
+
+        const auto * col_lon1 = checkAndGetColumn<ColumnFloat64>(arguments[1].column.get());
+        if (!col_lon1)
+            throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "Illegal type {} of argument {} of function {}. Must be Float64",
+                    arguments[1].type->getName(),
+                    2,
+                    getName());
+        const auto & data_lon1 = col_lon1->getData();
+
+        const auto * col_lat2 = checkAndGetColumn<ColumnFloat64>(arguments[2].column.get());
+        if (!col_lat2)
+            throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "Illegal type {} of argument {} of function {}. Must be Float64",
+                    arguments[2].type->getName(),
+                    3,
+                    getName());
+        const auto & data_lat2 = col_lat2->getData();
+
+        const auto * col_lon2 = checkAndGetColumn<ColumnFloat64>(arguments[3].column.get());
+        if (!col_lon2)
+            throw Exception(
+                    ErrorCodes::ILLEGAL_COLUMN,
+                    "Illegal type {} of argument {} of function {}. Must be Float64",
+                    arguments[3].type->getName(),
+                    4,
+                    getName());
+        const auto & data_lon2 = col_lon2->getData();
+
+        auto dst = ColumnVector<Float64>::create();
+        auto & dst_data = dst->getData();
+        dst_data.resize(input_rows_count);
+
+        for (size_t row = 0; row < input_rows_count; ++row)
+        {
+            const double lat1 = data_lat1[row];
+            const double lon1 = data_lon1[row];
+            const auto lat2 = data_lat2[row];
+            const auto lon2 = data_lon2[row];
+
+            LatLng point1 = {degsToRads(lat1), degsToRads(lon1)};
+            LatLng point2 = {degsToRads(lat2), degsToRads(lon2)};
+
+            Float64 res = distanceRads(&point1, &point2);
```

Comment:
Difference between h3PointDistRads, h3PointDistM, h3PointDistKm only in this single line. Lets make template class and just pass this function as template argument.